etcd/cluster-etcd-operator.md

```diff
@@ -0,0 +1,331 @@
+    managementState: Managed
+    observedConfig:
+      cluster:
+        members:
+        - name: etcd-member-ip-10-0-154-109.us-east-2.compute.internal
+          peerURLs: https://etcd-0.sbatsche.devcluster.openshift.com:2380
+          status: Ready
+        - name: etcd-member-ip-10-0-164-65.us-east-2.compute.internal
+          peerURLs: https://etcd-2.sbatsche.devcluster.openshift.com:2380
+          status: Ready
+        pending:
+        - name: etcd-member-ip-10-0-135-168.us-east-2.compute.internal
+          peerURLs: https://etcd-1.sbatsche.devcluster.openshift.com:2380
+          status: Remove
+```
+MemberCondition Matrix:
+
+| Condition | Bucket | Status | Result |
+| :---         |     :---:      |     :---:      | :---          |
+| etcd-member container is in CrashLoopBackoff | pending | Remove | Scale down member, stop Pod, remove data-dir, scale up member, and start Pod |
+| etcd-metric container is in CrashLoopBackoff | member | Degraded | Restart Pod |
+| Node reboots | pending | Unknown | No scaling action takes place |
+| Node in network partition | pending | Unknown | No scaling action takes place |
+| New static Pod starts for first time | pending | Add | Scale member up |
+| Previously CrashLoop Pod is started | pending | Unknown | After the Pod is observed as not actively in CrashLoopBackoff status changes to Add |
+| Node removed | pending | Remove | Scale member down |
+| Node added | pending | Add | Scale member up |
+| All containers Ready | member | Ready | No scaling action takes place |
+
+### User Stories
+
+#### Story 1
+
+User installs cluster and after a period of time a reboot on a node causes catastrophic failure resulting in etcd-member container crashlooping. Cluster etcd operator observes this failure and replaces failed static Pod without human interaction.
+
+#### Story 2
+
+User after install of cluster needs to replace Node because of hardware failure. When Node is removed operator scales down etcd cluster. When new Node is added to cluster, the operator performs scale up process adding the etcd member to the cluster.
+
+#### Story 3
+
+User would like to scale to 5 node etcd cluster from 3. User adds nodes to cluster and operator scales etcd accordingly.
+
+### Implementation Details/Notes/Constraints
+
+#### Installer
+
+During install we provision a single member etcd cluster on the bootstrap node. This etcd instance allows for the new master node apiservers to connect the bootstrap etcd endpoint early and allow CVO to be deployed much faster than the current implementation. Instead of waiting for quorum of a 3 node etcd cluster using SRV discovery to bootstrap we have quorum of 1 out of the gate. As etcd is now managed by the operator the cluster-etcd-operator will reconcile state and scale up to an eventual 4 node etcd cluster (bootstrap plus 3 masters).  At that point, the operator will remove the bootstrap etcd instance completing the bootstrap process scaling the cluster down to 3.
```

Comment:
Right I think that should just work, we can wait until SRV is populated to progress on the node level for your case instead of assuming these records already exist. I believe in this way we can possibly eliminate the need for waiting on bootstrap node as originally proposed.